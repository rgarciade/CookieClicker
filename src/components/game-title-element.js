import { LitElement, html } from 'lit';
import './icon-element.js';
import { gameTitleStyle } from './styles/game-title-style.js';

export class GameTitleElement extends LitElement {
	static get styles() {
		return [gameTitleStyle];
	}
	/**
	 * @property {backgroundColor} width - Background color of the button.
	 */
	static get properties() {
		return {
			backgroundColor: { type: String },
		};
	}

	get GameTitleBackgroundColor() {
		return this.backgroundColor
			? `--background-color: ${this.backgroundColor};`
			: '';
	}

	render() {
		return html`
			<style>
				:host {
				   ${this.GameTitleBackgroundColor}
				}
			</style>
			<slot></slot>
		`;
	}
}

window.customElements.define('game-title-element', GameTitleElement);
