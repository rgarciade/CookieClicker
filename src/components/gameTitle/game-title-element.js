import { LitElement, html } from 'lit';
import '../icon/icon-element.js';
import { gameTitleStyle } from './game-title-style.js';

export class GameTitleElement extends LitElement {
	static get styles() {
		return [gameTitleStyle];
	}

	render() {
		return html` <slot></slot> `;
	}
}

window.customElements.define('game-title-element', GameTitleElement);
