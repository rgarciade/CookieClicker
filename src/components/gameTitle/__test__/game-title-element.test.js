import { html, fixture } from '@open-wc/testing';
import '../game-title-element.js';
import { getAllElementsAndShadowRoots } from 'shadow-dom-testing-library';
import { expect } from '@jest/globals';

describe('game-title-element', () => {
	it('matches the snapshot', async () => {
		const el = await fixture(html`<game-title-element></game-title-element>`);
		const data = getAllElementsAndShadowRoots(el, '*');
		expect(data).toMatchSnapshot();
	});
});
