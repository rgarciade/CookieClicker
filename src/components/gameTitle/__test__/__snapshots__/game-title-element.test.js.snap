// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game-title-element matches the snapshot 1`] = `
[
  <game-title-element />,
  ShadowRoot {},
  <slot />,
  <style>
    
	:host {
		font-size: var(--title-font-size, 70px);
		color: var(--text-color, black);
		display: flex;
		gap: 40px;
		justify-content: flex-start;

		background-color: var(--background-color, white);
	}
	.game-title :last-child {
		margin-left: auto;
		padding-right: 20px;
	}
	.game-title :first-child {
		padding-left: 20px;
	}

	:host ::slotted(*) {
		display: flex;
		justify-content: flex-start;
		align-items: center;
		line-height: 0.2;
	}

	:host ::slotted(*:last-child) {
		margin-left: auto;
		padding-right: 40px;
	}
	:host ::slotted(*:first-child) {
		padding-left: 40px;
	}

  </style>,
]
`;
